docs/examples/tests/test_migrating_from_quantum_engine.py

```diff
@@ -0,0 +1,224 @@
+        from orquestra import sdk
+
+        @sdk.task(
+            source_import=sdk.GithubImport(
+                "zapatacomputing/my_source_repo",
+                git_ref="feat/my-feature-branch",
+            )
+        )
+        def my_task():
+            ...
+
+        # </snippet>
+        return my_task
+
+    @staticmethod
+    def github_import_private_before():
+        from orquestra import sdk
+
+        @sdk.task(
+            source_import=sdk.GithubImport(
+                "zapatacomputing/my_source_repo",
+                git_ref="feat/my-feature-branch",
+            )
+        )
+        def demo_task():
+            pass
+
+        # </snippet>
+        return demo_task
+
+    @staticmethod
+    def github_import_private_after():
+        from orquestra import sdk
+
+        @sdk.task(
+            source_import=sdk.GithubImport(
+                "zapatacomputing/my_source_repo",
+                git_ref="feat/my-feature-branch",
+                username="my-github-username",
+                personal_access_token=sdk.Secret(
+                    "my_pat",
+                    config_name="prod-d",
+                    # Orquestra Workspace where the secret is defined. You can read it
+                    # on Orquestra Portal website.
+                    workspace_id="ws1",
+                ),
+            )
+        )
+        def demo_task():
+            pass
+
+        # </snippet>
+        return demo_task
+
+    @staticmethod
+    def git_import_with_auth():
+        from orquestra import sdk
+
+        @sdk.task(
+            source_import=sdk.GitImportWithAuth(
+                repo_url="https://example.com/some_person/some_repo.git",
+                git_ref="v1.0.1",
+                username="my-git-username",
+                auth_secret=sdk.Secret(
+                    "access-token",
+                    config_name="prod-d",
+                    # Orquestra Workspace where the secret is defined. You can read it
+                    # on Orquestra Portal website.
+                    workspace_id="ws1",
+                ),
+            )
+        )
+        def demo_task():
+            pass
+
+        # </snippet>
+        return demo_task
+
+
+def _import_models(
+    dsl_task,
+) -> t.Tuple[ir.Import, t.Optional[t.Sequence[ir.Import]]]:
+    """
+    Uses the ``dsl_task`` exactly once in a workflow, builds the workflow model, and
+    extracts imports used by the corresponding task def model.
+    """
+    # TODO: reuse across test files
+
+    @sdk.workflow
+    def wf():
+        return dsl_task()
+
+    wf_model = wf().model
+
+    task_model = list(wf_model.tasks.values())[0]
+    src_import = wf_model.imports[task_model.source_import_id]
+    if (deps_ids := task_model.dependency_import_ids) is not None:
+        deps_imports = [wf_model.imports[dep_id] for dep_id in deps_ids]
+    else:
+        deps_imports = None
+
+    return src_import, deps_imports
+
+
+@pytest.mark.slow
```

Comment:
Question, why are those tests slow? I dont see them using like Ray or running any workflows in particular

Reply:
Thanks, fixed